### ISSUE TITLE
symbols with `keyof` are not renamed in JSDoc comments.

### ISSUE THREAD

**Issue by jespertheend** [microsoft/TypeScript#51889]
# Bug Report

### 🔎 Search Terms

keyof rename jsdoc 

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about `rename`

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?filetype=js#code/PTAEAUFcBdQTwPaQE6gMYoM4NQCwKbL4A0oADkZpqAGIBMoAhgHYAmoRzjAtvgFAhQQ4QB8BYYUIBqfNAmaZY3OFMaoAvKADefSQDMECAFygAzMV3CARmpMAWCwF8A3HwEAqd6AAC0OGXxWfD1tAGt8RBDlVWRHUABZOAAVf3xQd2AgA)

### 💻 Code

```ts
// Put your cursor here, press F2 and rename
//      |
//      V
const myVar = {
    foo: 3,
    bar: 4,
};

/** @typedef {keyof myVar} MyType */
```

### 🙁 Actual behavior

only `const myVar` was renamed, the `@typedef` didn't and as a result started emitting a type error.

### 🙂 Expected behavior

I'm not sure what the expected behavior should be, either:
 - `keyof myVar` should get renamed in the jsdoc comment, or
 - `keyof myVar` shouldn't be possible without a type error in the first place. For example, in TypeScript without JSDoc, this example already emits a type error without renaming, as can be seen in [this playground example](https://www.typescriptlang.org/play?#code/MYewdgzgLgBAtgTwGoEMBOMC8MDeAoGQmAMxBAC4YBmAGgKICN1KAWOgXwG488oEAHAKYwAsggAqA4dgDWghCGLxk6bgHo1RLdp0A9ffoC0xmAHJEqNKZhpBxQWggwoIGChgA3FABsAroJoYBl9YAEsnBkFQsABzGF8IQQATNwg8DR1MrJ13PiEYAAsHQQA6GAARUJSFX3hBFDAzPMFFZUtTAH4gA):
```ts
const myVar = {
    foo: 3,
    bar: 4,
};

type MyType = keyof myVar;
//                  ^^^^^--- 'myVar' refers to a value, but is being used as
//                           a type here. Did you mean 'typeof myVar'?
```

**Comment by fatcerberus** [comment]
You probably want to write `keyof typeof myVar`.  If you do then it gets renamed properly.  As you pointed out, `keyof myVar` isn't legal TS, so IntelliSense doesn't recognize it as referring to your variable--it's just nonsense.

**Comment by jespertheend** [comment]
Yeah I noticed that as well. It appears to be legal in JSDoc comments though, not sure if that's intentional. But if it is, I'd expect it to get renamed correctly. If it isn't intentional I'd expect it to give a type error just like in TypeScript.

**Comment by RyanCavanaugh** [comment]
Being able to use values as their corresponding `typeof`s is very much intentional

**Comment by RyanCavanaugh** [comment]
Closing language service bugs related to the 6.0 implementation. For more information, see #62827
